lib/Conversion/TTKernelToEmitC/TTKernelToEmitC.cpp

```diff
@@ -137,6 +145,10 @@ class TTMetalToEmitCFuncArgsRewriter
     rewriter.startOpModification(op);
     rewriter.setInsertionPointToStart(&op.getCallableRegion()->front());
     for (auto arg : blockArgs) {
+      // Skip initialization if the argument is not a CBType (SemaphoreType)
+      if (!mlir::isa<ttkernel::CBType>(arg.getType())) {
+        continue;
+      }
```

Comment:
Are there any constraints on where semaphore args are allowed to appear?  There might be some places in the code that assume a 1-1 mapping of cb block arg to top level op arg index.

Not suggesting this code is an issue, just wondering.

Reply:
For now I have added it after CB arguments. A lot of our code is written in a way to expect that first few arguments are CBs, and expect 1-1 mapping, so adding semaphores after it is not breaking anything.

Reply:
Should we make this part of the op verifier to enforce it?